source/vir/src/func_to_air.rs

```diff
@@ -877,3 +894,31 @@ pub fn func_def_to_air(
         }
     }
 }
+
+fn map_expr_rename_vars(
+    e: &Arc<SpannedTyped<crate::ast::ExprX>>,
+    req_ens_e_rename: &HashMap<Arc<String>, Arc<String>>,
+) -> Result<Arc<SpannedTyped<crate::ast::ExprX>>, Arc<air::messages::MessageX>> {
+    ast_visitor::map_expr_visitor_env(
+        e,
+        &mut air::scope_map::ScopeMap::new(),
+        &mut (),
+        &|_state, _, expr| {
```

Comment:
Should we worry about variable shadowing like this:
```
        &|_state, map, expr| {
            use crate::ast::ExprX;
            Ok(match &expr.x {
                ExprX::Var(i) if !map.contains_key(i) => {
                    expr.new_x(ExprX::Var(req_ens_e_rename.get(i).unwrap_or(i).clone()))
                }
                ExprX::VarLoc(i) if !map.contains_key(i) => {
                    expr.new_x(ExprX::VarLoc(req_ens_e_rename.get(i).unwrap_or(i).clone()))
                }
                ExprX::VarAt(i, at) if !map.contains_key(i) => {
                    expr.new_x(ExprX::VarAt(req_ens_e_rename.get(i).unwrap_or(i).clone(), *at))
                }
                _ => expr.clone(),
            })
        },
```
or should we just stop worrying and let `ItemLocalId` and `unique_local_name` take care of scoping?